modules/cnf-assigning-a-sriov-network-to-a-vrf.adoc

```diff
@@ -0,0 +1,81 @@
+[id="cnf-assigning-a-sriov-network-to-a-vrf_{context}"]
+= Assigning an SR-IOV network to a VRF
+
+[IMPORTANT]
+====
+The VRF feature described here is for Tech Preview purposes only and is not supported by Red Hat at this time.
+====
+
+As a cluster administrator, you can assign an SR-IOV network interface to your VRF domain by using the VRF CNI plug-in.
+
+To do this, you must add the VRF configuration to the optional `metaPlugins` parameter of the `SriovNetwork` resource.
+
+[id="cnf-creating-an-additional-sriov-network-with-vrf-plug-in_{context}"]
+== Creating an additional SR-IOV network attachment with the VRF CNI plug-in
+
+The SR-IOV Network Operator manages additional network definitions. When you specify an additional SR-IOV network to create, the SR-IOV Network Operator creates the NetworkAttachmentDefinition custom resource (CR) automatically.
+
+[NOTE]
+====
+Do not edit `NetworkAttachmentDefinition` CRs that the SR-IOV Network Operator manages. Doing so might disrupt network traffic on your additional network.
+====
+
+.Prerequisites
+* Install the {product-title} CLI (oc).
+* Log in to the {product-title} cluster as a user with cluster-admin privileges.
+
+.Procedure
+. Create the `SriovNetwork` CR by running the following command:
++
+[source,terminal]
+----
+$ oc edit sriovnetwork.openshift.io cluster
+----
+. Extend the CR that you are creating by adding the `metaPlugins` configuration for the additional network you are creating, as in the following example CR.
+. Save your changes and quit the text editor to commit your changes. The following YAML configures the VRF CNI plug-in:
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+  name: example-network
+  namespace: additional-sriov-network-1
+spec:
+  ipam: |
+    {
+      "type": "host-local",
+      "subnet": "10.56.217.0/24",
+      "rangeStart": "10.56.217.171",
+      "rangeEnd": "10.56.217.181",
+      "routes": [{
+        "dst": "0.0.0.0/0"
+      }],
+      "gateway": "10.56.217.1"
+    }
+  vlan: 0
+  resourceName: intelnics
+  metaPlugins : |
+    {
+      "type": "vrf", <1>
+      "vrfname": "example-vrf-name" <2>
+    }
+----
+<1> `type` must be set to `vrf`.
+<2> `vrfname` is the name of the VRF that the interface is assigned to. If it does not exist in the pod, it is created.
```

Comment:
Worth adding a note that this works only for resources of type `netdevice`

Reply:
Should we add `deviceType: netdevice` to the spec in the example?

Reply:
Nope, because that is driven by the nodepolicy that in turn is bound to the resource we are using.
I'd just drop a note

Reply:
OK. So note will say: "VRF plugins can only be specified for resources of type `netdevice`." ?

Reply:
"VRF will work only when the resource is of type `netdevice`".

Reply:
OK, adding that.